### PR TITLE
Fix PR monitor merge completion observability

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_b078879df0dd417784afb8b9` (agent: `codex`, model: `gpt-5.5`, effort: `xhigh`).

### Task
Fix PR monitor merge-completion observability and persistence discovered while monitoring PR #220.

Context: PR #220 merged successfully, but two small AWF issues were observed:
1. The built-in 90s pre-merge settle inside the merge critical section has weak logging; it looks frozen after `monitor.merge_critical_section_entered` until the next event.
2. `awf workspace show` reported `pr_merge_sha=null` even though monitor operations correctly recorded the merge SHA.

Requirements:
- Use strict TDD. Add failing regression tests first, then implement the smallest correct fix.
- Add explicit structured logs around pre-merge settle, e.g. started and elapsed/completed events. Include workspace id, PR number, base branch, head SHA, and wait seconds/elapsed seconds.
- Persist the successful merge SHA onto the workspace record on PR monitor completion so workspace detail/observability shows the merge SHA after auto-merge.
- Keep merge behavior unchanged. This is observability/persistence only.
- Use real tests, no fake Postgres assumptions. Preserve the 99% coverage standard.
- Validate with focused tests, ruff/mypy for touched code, then full coverage gate.

---
Validation: 5 profile command(s) passed inside the workspace container.

## Summary by CodeRabbit

* **New Features**
  * Added structured observability logging for pre-merge settle operations, including start/completion events with timing data
  * Improved merge SHA persistence when handling already-completed workspaces

* **Tests**
  * Added comprehensive unit tests for pre-merge settle logging events and merge SHA persistence in edge cases

### CODE REVIEW

**Comment by dimileeh** [comment]
⚠️ PR #221 needs human attention at commit `40a8d39c0a`.

AWF did not auto-merge because a review bot reported that review was skipped or left a trigger-review checklist unresolved.

After the blocker is cleared or a new commit lands, AWF will re-verify the PR before taking any merge action.

File: src/awf/runtime/pr_monitor_runner.py [hunk 1]
```diff
@@ -3015,6 +3038,32 @@ async def _record_merge_coordination_event(
...
+        wait_seconds: float,
+        elapsed_seconds: float | None = None,
+    ) -> None:
+        payload: dict[str, object] = {
+            "workspace_id": workspace_id,
+            "repo_url": repo_url,
+            "base_branch": base_branch,
+            "pr_number": pr_number,
+            "head_sha": status.head_sha,
+            "wait_seconds": wait_seconds,
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/pr_monitor_runner.py:3063]: fixed in commit c988db42eb326b0c74bc46dde1e44c7082e8021f

**Comment by dimileeh** [comment]
@coderabbitai review
